src/string-builder.js

```diff
@@ -52,4 +85,38 @@ StringBuilder.prototype.catIf = function(...args) {
     }
 
     return this;
-};
\ No newline at end of file
+};
+
+StringBuilder.prototype.wrap = function(prefix, suffix) {
+    wrap = new Decorator('wrap', prefix, suffix);
+    this.decorators.push(wrap);
+
+    return this;        
+};
+
+StringBuilder.prototype.prefix = function(...args) {
+    prefix = new Decorator('prefix', args);
+    this.decorators.push(prefix);
+
+    return this;
+};
+
+StringBuilder.prototype.suffix = function(...args) {
+    suffix = new Decorator('suffix', args);
+    this.decorators.push(suffix);
+
+    return this;
+};
+
+StringBuilder.prototype.end = function(deep) {
+    if (deep) {
+        for (let i=0; i < deep && this.decorators.length > 0; i++) {
+            this.decorators.pop();
+        }        
+    } else {
+        this.decorators = [];
```

Comment:
The `.end()` with no arguments is meant to remove the last decorator as `.end(1);`